edb/graphql/types.py

```diff
@@ -396,16 +397,26 @@ def graphql_schema(self) -> GraphQLSchema:
 
     def get_gql_name(self, name: s_name.QualName) -> str:
         module, shortname = name.module, name.name
+
+        # Adjust the shortname.
+        if shortname.startswith('__'):
+            # Use '_edb' prefix to mark derived and otherwise
+            # internal types. We opt out of '__edb' because we
+            # still rely on the first occurrence of '__' in
+            # GraphQL names to separate the module from the rest
+            # of the name in some code.
+            shortname = f'_edb{shortname}'
+        elif shortname.startswith('('):
```

Comment:
This is obviously kind of hacky. If it would be easy to actually be looking at the type instead of just the name, that would be an improvement, but if not then hacky isn't the end of the world.